tests/tests_unit/test_meta.py

```diff
@@ -19,10 +21,20 @@ def keep(path):
         ]
         return all(skip not in str(path.as_posix()) for skip in skip_list)
 
-    all_filepaths = Path("cognite/client/").glob("**/*.py")
     err_msg = "File: '{}' is missing 'from __future__ import annotations' at line=0"
-
-    for filepath in filter(keep, all_filepaths):
+    for filepath in filter(keep, ALL_FILEPATHS):
         with filepath.open("r") as file:
             # We just read the first line from each file:
             assert file.readline() == "from __future__ import annotations\n", err_msg.format(filepath)
+
+
+def test_ensure_all_tests_use_camel_case_except_dump():
+    err_msg = "File: '{}' contains camel_case=True and not a dump at line {}."
+    for filepath in ALL_FILEPATHS:
+        if "tests/" in str(filepath):
+            with filepath.open("r") as file:
+                for line_num, line in enumerate(file.readlines()):
+                    if ".dump(camel_case" in line:
```

Comment:
This would skip files with arguments splatted to multiple lines. I would attack this from either AST parsing, or using the inspect module on the class objects, something along these lines:
```py
from tests.utils import all_subclasses

for cls in all_subclasses(CogniteResource):
    if method := getattr(cls, "to_pandas"):
        if param := inspect.signature(method).parameters.get("camel_case"):
            assert param.default is False
```
+ add more base classes that we want to be easily dataframe-convertible.

Reply:
I guess we don't need `getattr` when we know all subclasses has this method... lol whoopsie. But for a general class, it would be beneficial ofc (with an additional third-argument `None` to not error when missing).

Reply:
Yeah, we talked about some of this offline and on Slack (although good to document here). Just a bit busy with customer stuff but will get to it soon.